controllers/overlaynetwork_controller.go

```diff
@@ -47,9 +53,21 @@ type OverlayNetworkReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *OverlayNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	overlayNetwork := &selfservicev1.OverlayNetwork{}
+	if err := r.Client.Get(ctx, req.NamespacedName, overlayNetwork); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
```

Comment:
If there is existing NAD with the same name we should raise a error.
We should differentiate between a NAD created by SelfOverlay and a NAD created by the admin.
It can be done using the owwer reference,

Reply:
Also if the OverlayNetwork has a DelertionTimestamp we should return.